yt/fields/derived_field.py

```diff
@@ -489,17 +489,34 @@ def __copy__(self):
 
 
 class FieldValidator:
+    """
+    Base class for FieldValidator objects. Available subclasses include:
+    """
+
+    # note: the above docstring is dynamically modified with a list of child
+    # classes by _set_field_validator_base_docstring() below.
     pass
 
 
 class ValidateParameter(FieldValidator):
-    def __init__(self, parameters, parameter_values=None):
-        """
-        This validator ensures that the dataset has a given parameter.
+    """
+    A :class:`FieldValidator` that ensures the dataset has a given parameter.
 
-        If *parameter_values* is supplied, this will also ensure that the field
+    Parameters
+    ----------
+    parameters: Union[str, Iterable[str]
```

Comment:
removed all the Union and Optional occurrences. left the iterable though? Should I remove that too? it is nice to note that it could be a str or any iterable with str's. but I could remove the iterable and rephrase the description